src/guides/protocol.md

```diff
@@ -0,0 +1,230 @@
+Every 36 blocks (about 6 hours)
+[the txn is committed to the urkel tree](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/blockchain/chaindb.js#L1842-L1843).
+
+From the whitepaper:
+
+> Because our tree is implemented as a series of append-only files, a
+> commission interval is required to prevent history bloat, which may otherwise
+> require the user of the software to compact their history regularly.
+
+### Auctions
+
+Name auctions progress through a [series of states based on time (i.e. block depth).](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/covenants/namestate.js#L127-L159)
+The first time a name is seen on chain, it's state is
+[set to the current block height.](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/blockchain/chain.js#L908)
+This height is saved in the database and used as a reference point for the entire
+auction process. It can be [reset](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/covenants/namestate.js#L229)
+after a name expires. Covenant types will be rejected if they are broadcast during the
+wrong phase.
+
+| Blocks since<br>previous state   | Approx. time | Current state     |
+|----------------------------------|--------------|-------------------|
+| (undefined name)                 |              | `CLOSED`          |
+| < 37                             | ~6 hours     | `OPEN`            |
+| < 720                            | ~5 days      | `BIDDING`         |
+| < 1,440                          | ~10 days     | `REVEAL`          |
+| > 0                              |              | `CLOSED`          |
+
+When reserved names are `CLAIM`ed on chain, they enter a
+[lockup period of about 30 days](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/networks.js#L243).
+During the `LOCKED` phase,
+[a reserved name can be re-claimed](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/blockchain/chain.js#L925-L936).
+This is to provide solutions to legacy DNS name holders whose keys are compromised.
+
+### Claims
+
+Legacy DNS names in the Alexa top 100,000 are reserved for their current owners.
+You can see the list in [names.json](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/covenants/names.json).
+
+[Learn more about the claims list.](https://github.com/handshake-org/hs-names)
+
+[Learn about on-chain name claims.](https://handshake-org.github.io/guides/claims.html)
+
+### Name rollout
+
+Un-reserved names are not all available immediately when mainnet launches. They
+become available over the first year, based on a simple algorithm that
+[compares the name's hash against a 52-week modulus.](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/covenants/rules.js#L274-L295)
```

Comment:
Maybe mention that you can use the RPC `getnameinfo` to determine when the name will be available

Reply:
Sure, wouldn't hurt. I just wanted to focus more abstractly on the protocol in this doc rather than usage, but I'll add this one.